upload/src/addons/SV/ThreadReplyBanner/XF/Pub/Controller/Thread.php

```diff
@@ -5,6 +5,69 @@
 /*
  * Extends \XF\Pub\Controller\Thread
  */
+use XF\Mvc\ParameterBag;
+use XF\Mvc\Reply\View;
+use XF\Repository\Notice;
+
 class Thread extends XFCP_Thread
 {
+	public function actionReplyBannerHistory(ParameterBag $params) {
+		return $this->rerouteController(
+			'XF:EditHistory', 'index',
+			[
+				'content_type' => 'thread_banner',
+				'content_id'   => $params->get('thread_id')
+			]
+		);
+	}
+
+	public function actionIndex(ParameterBag $params) {
+		$reply = parent::actionIndex($params);
+
+		if ($reply instanceof View) {
+			if ($reply->getParam('thread')->getRelation('ThreadBanner')) {
+				$notice = [
+					'title' => '',
+					'message' => $reply->getParam('thread')->getRelation('ThreadBanner')->getValue('raw_text'),
```

Comment:
The banner is bbcode text, and not raw html. This needs to be rendered first, which should be done in a method hosted in the ThreadBanner entity